### PR TITLE
Fix pagination beyond page 2

### PR DESCRIPTION
**Pull request by timo-schmid**
In the process of changing `Search.repositories` to be paginated (#30) I noticed that I never get results beyond page 2.

I discovered that the `Link` header contains multiple links separated by `,` in one header line.
For example
```zsh
curl -v https://api.github.com/search/repositories\?q\=scala\&sort\=stars\&page\=2 -O /dev/null
```
wiill return
```
Link: <https://api.github.com/search/repositories?q=scala&sort=stars&page=1>; rel="prev", <https://api.github.com/search/repositories?q=scala&sort=stars&page=3>; rel="next", <https://api.github.com/search/repositories?q=scala&sort=stars&page=34>; rel="last", <https://api.github.com/search/repositories?q=scala&sort=stars&page=1>; rel="first"
```
On the first page it works, because the first header there is the `rel="next"` link.

To fix it, I rewrote the logic for `getNextUri`.

The [spec](https://tools.ietf.org/html/rfc5988) (page 5) seems to indicate this is not a standard, but it appears in the examples (page 9). So maybe it should be fixed ustream in http4s. I'd like to hear your thoughts on that.

While fixing the above I also noticed that `runPaginatedRequest` seemed to ignore invalid HTTP status codes. This change is kind of optional because the parsing of the response will later fail anyway, but I think it's a bit nicer to check the status code of the response first and the error message is much clearer.

### CODE REVIEW

**Comment by ChristopherDavenport** [comment]
Lets get an issue on http4s, and make sure we're handling this where most appropriate and then coping with it here in the process.

File: core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala [hunk 1]
```diff
@@ -46,44 +47,49 @@ object RequestConstructor {
...
     val uri = Uri.resolve(baseUrl(auth), extendedUri)
-
-    unfoldLoopEval(uri){uri => 
+    unfoldLoopEval(uri){ uri =>
       val baseReq = Request[F](method = Method.GET, uri = uri)
         .withHeaders(extraHeaders)
       val req = auth.fold(baseReq)(setAuth(_)(baseReq))
-      c.run(req).use{resp => 
-          resp.as[B].map{
+      c.fetch(req) { resp =>
```
Comment (ChristopherDavenport) [inline core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala]: Why did we switch from `run` to `fetch`. `run` is clearer use semantics in my opinion.

Comment (timo-schmid) [reply core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala]: No clear reason - I just didn't know how to do it using f.
But looking at it now, I think it should also work using `use` as well.
I guess this part can be changed back - I can do it if you want.

**Comment by ChristopherDavenport** [comment]
I think this may be caught by https://github.com/http4s/http4s/pull/2967

**Pull request merged**